freshmaker/pyxis.py

```diff
@@ -0,0 +1,253 @@
+
+        if response.status_code == HTTPStatus.OK:
+            return response.json()
+
+        # Warn early, in case there is an error in the error handling code below
+        log.warning("Request to %s gave %r", response.request.url, response)
+
+        try:
+            response_text = response.json()
+        except ValueError:
+            response_text = response.text
+
+        raise PyxisRequestError(response.status_code, response_text)
+
+    def _pagination(self, entity, params):
+        """
+        Process all pages in Pyxis
+
+        :param str entity: what data/entity to request from Pyxis
+        :param dict params: parameters to add to GET request
+        :return: list of all 'data' fields from responses from Pyxis
+        :rtype: list
+        """
+        request_params = copy.deepcopy(params)
+        ret = []
+        page = 0
+        while True:
+            request_params["page"] = page
+            response_data = self._make_request(entity, params=request_params)
+            # When the page after the actual last page is released, data will be an empty list
+            if not response_data.get('data'):
+                break
+            ret.extend(response_data['data'])
+            page += 1
+
+        return ret
+
+    def get_operator_indices(self, params={}):
+        """
+        Get all index images for organization(s)(configurable) from Pyxis
+
+        :param dict params: parameters to add to GET request
+        """
+        request_params = copy.deepcopy(params)
+        organization = conf.pyxis_index_image_organization
+        request_params["filter"] = "organization==" + organization
+
+        entity = "operators/indices"
+        indices = self._pagination(entity, request_params)
+        return indices
+
+    def _get_bundles_per_index_image(self, index_images, params={}):
+        """
+        Get bundle images for all index images
+
+        :param list index_images: list of index images
+        :param dict params: parameters to add to GET request
+        :return: bundle images per index image
+        :rtype: dict
+        """
+        request_params = copy.deepcopy(params)
+        bundles_per_index_image = {}
+        # we need 'bundle_path_digest' to find ContainerImage of that bundle
+        include_fields = ['data.channel_name', 'data.version',
+                          'data.related_images', 'data.bundle_path_digest']
+        request_params['include'] = ','.join(include_fields)
+        for index_image in index_images:
+            path = index_image.get('path', '')
+            if not path:
+                continue
+            request_params['filter'] = f"source_index_container_path=={path}"
+            bundles = self._pagination('operators/bundles', request_params)
+            bundles_per_index_image[path] = bundles
+
+        return bundles_per_index_image
+
+    def get_bundles_per_channel_per_index_image(self, index_images, params={}):
```

Comment:
This makes it clearer what is returned
```suggestion
    def get_latest_bundles_per_channel_per_index_image(self, index_images, params={}):
```

Reply:
I just don't like too long name of methods =)

Reply:
@apaplaus yeah, that is rather long. How about `get_latest_bundles`? It's clear this is per index image based on the input parameters of `index_images`. The docstring would also clarify this.

Reply:
Okay maybe it's better, will change it.